src/thing/Portal.cpp

```diff
@@ -50,13 +57,44 @@ void Portal::Start() {
           }
         }
       }
+      function scan() {
+        document.getElementById('scanning').style.display='inline';
+        var xhr = new XMLHttpRequest();
+        xhr.open("GET", "/wifi/scan", true);
+        xhr.onreadystatechange = function () {
+          if (xhr.readyState==4 && xhr.status==200) {
+           load_networks(JSON.parse(xhr.responseText));
+           document.getElementById('scanning').style.display='none';
+          }
+        }
+        xhr.send();
+      }
+      function load_networks(networks) {
+        select = document.getElementById('networks');
+        select.innerHtml = '';
+        networks.sort(function(a, b) {
+          return b.rssi - a.rssi;
+        });
+        networks.forEach(function(network) {
+          option = document.createElement('option');
+          option.value = network.ssid;
+          option.text = network.ssid + ' (' + network.rssi +  ')';
+          select.add(option);
+        });
+        select.style.display='inline';
```

Comment:
something you could consider is setting a .class with the state ('scanning', 'selection', 'connected') on a parent div for the ui, and having some css that configure the visibility of all elements for each one instead of having the js show/hide them.